bears/xml/XMLBear.py

```diff
@@ -1,5 +1,26 @@
+import os
+import requests
+
 from coalib.bearlib.abstractions.Lint import Lint
 from coalib.bears.LocalBear import LocalBear
+from coalib.misc.Shell import escape_path_argument
+from coalib.settings.Setting import path
+
+
+def path_or_url(type1, type2):
+    '''
+    Checks if the argument is a ``url`` or a ``path``.
+
+    :param type1:       path
+    :param type2:       str
+    :return:            path or str
+    '''
+    def _path_or_url(dtdvalid):
+        if dtdvalid is not None and os.path.exists(dtdvalid):
+            return type1
```

Comment:
This doesnt look right
the function is supposed to return the converted value.

If I use `path_or_url(path, str)` right now, and the path is valid, you're returning `path`
While actually you should be returning the path-ified dtdvalue - i.e. `path(dtdvalue)`